modules/core/src/main/java/org/apache/ignite/internal/util/IgniteNameUtils.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.util;
+
+import static org.apache.ignite.lang.ErrorGroups.Common.ILLEGAL_ARGUMENT_ERR;
+
+import org.apache.ignite.lang.IgniteInternalException;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Utility methods used for cluster's named objects: schemas, tables, columns, indexes, etc.
+ */
+public final class IgniteNameUtils {
+    /** No instance methods. */
+    private IgniteNameUtils() {
+    }
+
+    /**
+     * Parse simple name: unquote name or cast to upper case not-quoted name.
+     *
+     * @param name String to parse object name.
+     * @return Unquoted name or name is cast to upper case. "tbl0" -&gt; "TBL0", "\"Tbl0\"" -&gt; "Tbl0".
+     */
+    public static String parseSimpleName(String name) {
+        if (StringUtils.nullOrEmpty(name)) {
```

Comment:
why do we allow empty or null names?

Reply:
This parser contains limited set of restrictions, because it is used for read-only operations on schema